tests/test_dicom/test_dicom_anonymise.py

```diff
@@ -1,7 +1,84 @@
+from os.path import abspath, basename, dirname, join as pjoin
+
+import pydicom
+from pymedphys.dicom import (
+    anonymise_dataset,
+    anonymise_directory,
+    anonymise_file,
+    BaselineDicomDictionary,
+    IDENTIFYING_KEYWORDS,
+    is_anonymised_dataset,
+    is_anonymised_directory,
+    is_anonymised_file
+)
+
+HERE = dirname(abspath(__file__))
+DATA_DIR = pjoin(dirname(HERE), 'data', 'dicom_anonymise')
+dicom_test_filepath = pjoin(DATA_DIR, "RP.almost_anonymised.dcm")
+
+
+def test_is_anonymised_dataset():
+    ds = pydicom.dcmread(dicom_test_filepath)
+
+    assert not is_anonymised_dataset(ds)
+
+    ds_anon_manual = ds
+    ds_anon_manual.PatientName = "Anonymous"
+
+    ds_anon_func = anonymise_dataset(ds, delete_private_tags=False,
+                                     copy_dataset=True)
+
+    # Patient Name should've been the only non-anonymous value. If not,
+    # test data has been changed
+    for elem_manual, elem_func in zip(ds_anon_manual, ds_anon_func):
+        assert elem_manual.value == elem_func.value
+
+    assert not is_anonymised_dataset(ds_anon_manual, ignore_private_tags=False)
+    assert not is_anonymised_dataset(ds_anon_func, ignore_private_tags=False)
+    assert is_anonymised_dataset(ds_anon_manual, ignore_private_tags=True)
+    assert is_anonymised_dataset(ds_anon_func, ignore_private_tags=True)
+
+    ds_anon_func_full = anonymise_dataset(ds, delete_private_tags=True)
+    assert is_anonymised_dataset(ds_anon_func_full, ignore_private_tags=False)
+
+
+def test_is_anonymised_file():
+    assert not is_anonymised_file(dicom_test_filepath)
+
+    # anonymous_filepath = pjoin(DATA_DIR, "RP.almost_anonymised.dcm")
+
+    dicom_anon_filepath = anonymise_file(dicom_test_filepath,
+                                         delete_private_tags=False)
+
+    assert not is_anonymised_file(dicom_anon_filepath,
+                                  ignore_private_tags=False)
+    assert is_anonymised_file(dicom_anon_filepath,
+                              ignore_private_tags=True)
+
+    dicom_anon_filepath = anonymise_file(dicom_test_filepath,
+                                         delete_private_tags=True)
+    assert is_anonymised_file(dicom_anon_filepath,
+                              ignore_private_tags=False)
+
+
+def test_is_anonymised_directory():
+    dirpath_expected_to_fail = DATA_DIR
+    assert not is_anonymised_directory(dirpath_expected_to_fail)
+
+    anonymous_filepath = anonymise_file(dicom_test_filepath)
+    dirpath_expected_to_pass = pjoin(DATA_DIR, 'Anonymous')
+    anonymous_filepath_moved = pjoin(dirpath_expected_to_pass,
+                                     basename(anonymous_filepath))
+
+    makedirs(dirpath_expected_to_pass, exist_ok=True)
+    replace(anonymous_filepath, anonymous_filepath_moved)
+
+    assert is_anonymised_directory(dirpath_expected_to_pass)
+
+    remove(anonymous_filepath_moved)
+    rmdir(dirpath_expected_to_pass)
```

Comment:
When tox is run in parallel mode across multiple virtual environments (as in testing against py36 and py37 simultaneously) it struggles with tests of this form. Reason being, one test is creating the same files that another is trying to delete etc.

To circumvent that I add a random string to the end of the filenames. See here:

https://github.com/pymedphys/pymedphys/blob/master/tests/test_dicom/test_header_tweaks.py#L40-L44

Another point, is if this code does error for whatever reason, the `remove` and `rmdir` commands are not run. Would you be able to use the `try`/`finally` pattern here to make sure that the directory and files are always removed even if the test fails?

See: https://github.com/pymedphys/pymedphys/blob/master/tests/test_dicom/test_header_tweaks.py#L47-L58

Reply:
Add the try/finally. But I might handle the parallel issue in another branch, if that's okay? It would actually require a bit of a rework to do properly...

Reply:
Actually, that latest commit achieves most of the safety you're after, I think. Still not perfect, but very close!

Reply:
> But I might handle the parallel issue in another branch, if that's okay? 

Yup, certainly okay